presto-spi/src/main/java/io/prestosql/spi/connector/ColumnMetadata.java

```diff
@@ -30,6 +30,7 @@
     private final String name;
     private final Type type;
     private final boolean nullable;
+    private final boolean hasDefault;
```

Comment:
This is a wrong abstraction. 

Think about a PostrgeSQL connector and PostrgeSQL table:

```
create table t (
   a bigint,
   b bigint,
   c bigint,
   d bigint default some_postgres_udf());
```

and users run this in Presto:

```
INSERT INTO t(a, b) VALUES (42, 13);
```

Currently Presto creates temporary table to hold the data being inserted.

- the temporary table needs to have `a`, `b` columns to hold the values 
- the temporary table must not have `d` column so that we can leverage PostgreSQL's calculating `some_postgres_udf()`
- the temporary table should not have `c` column, so that we don't write NULLs unnecessarily and the temporary table, inflating the data to be copied at the commit phase.

Thus, it doesn't matter whether column has `DEFAULT` clause or not.
What matters is whether user mentioned the column in their INSERT.

Of course, this is not so simple for Hive connector. In  that case we're responsible for storage ourselves.
We need to fill NULLs for columns not being explicitly INSERTed.
To be backward compatible, we should be able to do this on connector's behalf in the engine.
I don't know yet how exactly this should be addressed.